src/NServiceBus.Gateway/Routing/RouteToGatewayBehaviour.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using Gateway.Routing;
+    using Pipeline;
+    using Routing;
+    using TransportDispatch;
+
+    class RouteToGatewayBehaviour : Behavior<IRoutingContext>
+    {
+        public RouteToGatewayBehaviour(string gatewayAddress)
+        {
+            this.gatewayAddress = gatewayAddress;
+        }
+        
+        public override Task Invoke(IRoutingContext context, Func<Task> next)
+        {
+            RouteThroughGateway routeThroughGateway;
+
+            if (context.Extensions.TryGet(out routeThroughGateway))
+            {
+                //Hack 133
+                context.RoutingStrategies = new[]
+                {
+                    new UnicastRoutingStrategy(gatewayAddress)
+                };
+
+                context.Extensions.Remove<RouteThroughGateway>();
+            }
+
+            return next();
+        }
+
+        string gatewayAddress;
+
+        public class Registration : RegisterStep
```

Comment:
Accept in ctor here `gatewayAddress` and pass it on to the `RouteToGatewayBehaviour ` created in the func of the `RegisterStep`